ui/component/textareaWithSuggestions/view.jsx

```diff
@@ -0,0 +1,387 @@
+    } else {
+      inputRef.current.removeAttribute('term');
+    }
+  }, [inputRef, isLivestream, suggestionTerm]);
+
+  // Only resolve commentors on Livestreams when first trying to mention/looking for it
+  React.useEffect(() => {
+    if (isLivestream && commentorUris && suggestionTerm) doResolveUris(commentorUris);
+  }, [commentorUris, doResolveUris, isLivestream, suggestionTerm]);
+
+  // Allow selecting with TAB key
+  React.useEffect(() => {
+    if (!suggestionTerm) return; // only if there is a term, or else can't tab to navigate page
+
+    function handleKeyDown(e: SyntheticKeyboardEvent<*>) {
+      const { keyCode } = e;
+
+      if (highlightedSuggestion && keyCode === KEYCODES.TAB) {
+        e.preventDefault();
+        handleSelect(highlightedSuggestion.label);
+      }
+    }
+
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => window.removeEventListener('keydown', handleKeyDown);
+  }, [handleSelect, highlightedSuggestion, suggestionTerm]);
+
+  /** ------ **/
+  /** Render **/
+  /** ------ **/
+
+  const renderGroup = (group: string, children: any) => (
+    <div className="textareaSuggestions__group">
+      <label className="textareaSuggestions__label">
+        {group}
+        {suggestionTerm && suggestionTerm.length > 1
+          ? ' ' + __('matching %matching_term%', { matching_term: suggestionTerm })
+          : ''}
+      </label>
+      {children}
+      <hr className="textareaSuggestions__topSeparator" />
+    </div>
+  );
+
+  const renderInput = (params: any) => {
+    const { InputProps, disabled, fullWidth, id, inputProps: autocompleteInputProps } = params;
+    const inputProps = { ...autocompleteInputProps, ...inputDefaultProps };
+    const autocompleteProps = { InputProps, disabled, fullWidth, id, inputProps };
+
+    return <TextField inputRef={inputRef} multiline select={false} {...autocompleteProps} />;
+  };
+
+  const renderOption = (optionProps: any, label: string) => {
+    const emoteFound = isEmote && EMOTES.find(({ name }) => name.toLowerCase() === label);
+    const emoteValue = emoteFound ? { name: label, url: emoteFound.url } : undefined;
+    const emojiFound = isEmote && EMOJIS.getUnicode(label);
+    const emojiValue = emojiFound ? { name: label, unicode: emojiFound } : undefined;
+
+    return <TextareaSuggestionsItem uri={label} emote={emoteValue || emojiValue} {...optionProps} />;
+  };
+
+  return (
+    <Autocomplete
+      autoHighlight
+      disableClearable
+      filterOptions={(options) => options.filter(({ label }) => allMatches && allMatches.includes(label))}
+      freeSolo
+      fullWidth
+      getOptionLabel={(option) => option.label}
+      groupBy={(option) => option.group}
+      id={id}
+      inputValue={messageValue}
+      loading={!allMatches || allMatches.length === 0 || showPlaceholder}
+      loadingText={results || showPlaceholder ? __('Searching...') : __('Nothing found')}
+      onBlur={() => onBlur && onBlur()}
```

Comment:
In this video, the search results came back, but the gui is stuck at "Searching..."

https://www.notion.so/Temp-Links-31f1387c619d4f338b877c41b99fd59b#b1662ab22da947acbdd79f7350f8a08a